application/workspace/models.py

```diff
@@ -53,18 +66,53 @@ class WorkspaceInvitation(Base, Persistent):
     start_date = db.Column(db.Date, nullable=True)
 
 
-class HolidayCalendar(Base, Persistent):
-    __tablename__ = 'holiday_calendar'
+class WorkspaceHoliday(Base, Persistent):
+    __tablename__ = 'workspace_holiday'
 
     id = db.Column(db.Integer, primary_key=True)
     name = db.Column(db.String(255), nullable=False)
     ws_id = db.Column(db.Integer, db.ForeignKey('workspace.id'))
+    date = db.Column(db.Date, nullable=False)
 
+    @staticmethod
+    def get_work_days_count(ws_id, start_date, end_date):
+        holidays = WorkspaceHoliday.query().\
+            filter(WorkspaceHoliday.date.between(start_date, end_date)).\
+            filter(WorkspaceHoliday.ws_id == ws_id).\
+            all()
+        result = 0
+        holiday_dates = set(map(lambda h: h.date, holidays))
```

Comment:
you can use `frozenset` to make it explicit that the whole object is immutable

Reply:
Needed `set` to deduplicate, but it's no longer needed.